WordPress/Classes/ViewRelated/Me/Me Main/MeViewController+UIViewControllerRestoration.swift

```diff
@@ -0,0 +1,36 @@
+/// State Restoration
+extension MeViewController: UIViewControllerRestoration {
+
+    static func configureRestoration(on instance: MeViewController) {
+        instance.restorationIdentifier = "WPMeRestorationID"
+        instance.restorationClass = MeViewController.self
+    }
+
+    static func viewController(withRestorationIdentifierPath identifierComponents: [String],
+                               coder: NSCoder) -> UIViewController? {
+        if FeatureFlag.meMove.enabled {
+            let meController = MeViewController()
+            return meController
+        } else {
+            return WPTabBarController.sharedInstance().meViewController
+        }
+    }
+
+    override func decodeRestorableState(with coder: NSCoder) {
+        super.decodeRestorableState(with: coder)
+        if FeatureFlag.meMove.enabled {
+            // needs to be done after self has been initialized, so we do it in this method
+            let doneButton = UIBarButtonItem(target: self, action: #selector(dismissHandler))
+            navigationItem.rightBarButtonItem = doneButton
+            if WPDeviceIdentification.isiPad() {
+                navigationController?.modalPresentationStyle = .formSheet
+                navigationController?.modalTransitionStyle = .coverVertical
+            }
```

Comment:
From what I can tell, this check isn't necessary since `.formSheet` automatically adjusts depending on the trait environment.

> In a vertically regular, but horizontally compact environment, this option displays a sheet interface, where a portion of the underlying content remains visible near the top of the screen.
> 
> In a vertically compact environment, this option is essentially the same as UIModalPresentationStyle.fullScreen

This seems to mean Portrait iPhone would get the `.pageSheet` interface (the existing default for `.automatic`) while Landscape iPhone gets `.fullscreen`.

## After

<a href="https://user-images.githubusercontent.com/3250/75374692-fad13580-5889-11ea-8acd-76fcb0415758.png"><img src="https://user-images.githubusercontent.com/3250/75374692-fad13580-5889-11ea-8acd-76fcb0415758.png" width="35%"></a> <a href="https://user-images.githubusercontent.com/3250/75374700-fdcc2600-5889-11ea-8a41-1f36d3df6a08.png"><img src="https://user-images.githubusercontent.com/3250/75374700-fdcc2600-5889-11ea-8a41-1f36d3df6a08.png" width="60%"></a>


```suggestion
            navigationController.modalPresentationStyle = .formSheet
            navigationController.modalTransitionStyle = .coverVertical
```

Reply:
> From what I can tell, this check isn't necessary since `.formSheet` automatically adjusts depending on the trait environment.
> 
> > In a vertically regular, but horizontally compact environment, this option displays a sheet interface, where a portion of the underlying content remains visible near the top of the screen.
> > In a vertically compact environment, this option is essentially the same as UIModalPresentationStyle.fullScreen
> 
> ###### from [`UIViewController.modalPresentationStyle`](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621355-modalpresentationstyle#)
> This seems to mean Portrait iPhone would get the `.pageSheet` interface (the existing default for `.automatic`) while Landscape iPhone gets `.fullscreen`.
> 
> ## After
> <img alt="" width="35%" src="https://user-images.githubusercontent.com/3250/75374692-fad13580-5889-11ea-8acd-76fcb0415758.png"> <img alt="" width="60%" src="https://user-images.githubusercontent.com/3250/75374700-fdcc2600-5889-11ea-8a41-1f36d3df6a08.png">

This seems to be working as expected in iOS 13, but not on older version (Apple documentation should highlight the difference..). Here are a screenshots on iPhone 8 Plus, iOS 11.4 and 12.4
# 
![Screen Shot 2020-02-26 at 3 05 10 PM](https://user-images.githubusercontent.com/34376330/75388473-b1451280-58aa-11ea-8b28-a12270f2e19b.png)
# 
![Screen Shot 2020-02-26 at 3 05 24 PM](https://user-images.githubusercontent.com/34376330/75388484-b609c680-58aa-11ea-9095-457523d8c9ea.png)

Reply:
Yep! That definitely looks like the case. 😞